syntax = "proto3";

package flowpipe.stages.parquet.arrow.sink.v1;

// Configuration for ParquetArrowSink stage
message ParquetArrowSinkConfig {
  enum FileSystem {
    FILE_SYSTEM_AUTO = 0;
    FILE_SYSTEM_LOCAL = 1;
    FILE_SYSTEM_S3 = 2;
    FILE_SYSTEM_GCS = 3;
    FILE_SYSTEM_HDFS = 4;
  }

  enum Compression {
    COMPRESSION_UNCOMPRESSED = 0;
    COMPRESSION_SNAPPY = 1;
    COMPRESSION_GZIP = 2;
    COMPRESSION_BROTLI = 3;
    COMPRESSION_ZSTD = 4;
    COMPRESSION_LZ4 = 5;
    COMPRESSION_LZ4_FRAME = 6;
    COMPRESSION_LZO = 7;
    COMPRESSION_BZ2 = 8;
  }

  // File path to write
  string path = 1;

  // Row group size for parquet writer
  optional int64 row_group_size = 2;

  // Filesystem selection
  FileSystem filesystem = 3;

  // Compression selection
  Compression compression = 4;
}
